### PR TITLE
[payment-request] constructor() and show() tests

### PR DESCRIPTION
**Pull request by stan-stripe**
Add a test to the payment-request directory that tests that the acceptPromise is created and returned when show() is called on a PaymentRequest.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision 62e45ff4ede58d584b645e65016447e625d5474e
Using browser at version 58.0.3004.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4768/payment-request/payment-request-show-accept-promise-returned.html">/payment-request/payment-request-show-accept-promise-returned.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | TIMEOUT |

</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision 62e45ff4ede58d584b645e65016447e625d5474e
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4768/payment-request/payment-request-show-accept-promise-returned.html">/payment-request/payment-request-show-accept-promise-returned.html</a></summary>


| Subtest | Results |
|---------|---------|
|         | TIMEOUT |

</details>

**Comment by sideshowbarker** [comment]
w3c-test:mirror

File: payment-request/payment-request-show-accept-promise-returned.html
Comment (sideshowbarker) [inline payment-request/payment-request-show-accept-promise-returned.html]: You’re missing a needed `</script>` end tag here.

File: payment-request/payment-request-show-accept-promise-returned.html
Comment (sideshowbarker) [inline payment-request/payment-request-show-accept-promise-returned.html]: The`PaymentRequest` constructor requires at least two separate arguments: a _methodData_ argument and a _details_ argument, but this is feeding it a single array as an argument.

**Review by sideshowbarker — CHANGES_REQUESTED** [review]

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/4768)

**Comment by stan-stripe** [comment]
@sideshowbarker PTAL

Are the tests idiomatic? Happy to use other constructs as adequate.

**Comment by stan-stripe** [comment]
These tests pass using the reference implementation as Chrome extension available here: https://github.com/stan-stripe/js-payment-request

**Comment by ianbjacobs** [comment]
Hi gents, I was just checking to see how this is going! I see the tests are available on w3c-test.org; that sounds cool.

**Comment by stan-stripe** [comment]
@ianbjacobs waiting on review from @sideshowbarker at this stage

**Comment by sideshowbarker** [comment]
I will try to get the review on this done today

**Comment by sideshowbarker** [comment]
Do you know why the `frame` and `object` and `embed` tests in https://w3c-test.org/submissions/4768/payment-request/allowpaymentrequest/allowpaymentrequest-attribute-same-origin-bc-containers.https.html are failing?

**Comment by sideshowbarker** [comment]
All these have tests that are failing. They’re all throwing.

* https://w3c-test.org/submissions/4768/payment-request/allowpaymentrequest/allowpaymentrequest-attribute-same-origin-bc-containers.https.html
* https://w3c-test.org/submissions/4768/payment-request/allowpaymentrequest/no-attribute-same-origin-bc-containers.https.html
* https://w3c-test.org/submissions/4768/payment-request/allowpaymentrequest/removing-allowpaymentrequest.https.sub.html
* https://w3c-test.org/submissions/4768/payment-request/allowpaymentrequest/setting-allowpaymentrequest-timing.https.sub.html
* https://w3c-test.org/submissions/4768/payment-request/allowpaymentrequest/setting-allowpaymentrequest.https.sub.html

I’ve not had time to look at them in detail yet to determine why.

**Review by sideshowbarker — DISMISSED** [review]

**Comment by sideshowbarker** [comment]
Please change the `echo-PaymentRequest.html` filename to `echo-PaymentRequest-support.html`.

That will cause the manifest generator to recognize it as a support file rather than treating it as a test.

**Comment by marcoscaceres** [comment]
@stan-stripe OMG, I'm so so so terribly sorry! I just found these tests. We've added a bunch of new tests that cover these exact cases. Darn :( this could have saved us some time and duplication.

@stan-stripe Would you like to be part of the testing effort still? We really need more reviewers and people to help us with additional tests.

**Pull request closed without merging**
